pkg/client/didexchange/client.go

```diff
@@ -62,6 +63,22 @@ func (c *Client) CreateInvitation() (*InvitationRequest, error) {
 	}}, nil
 }
 
+// HandleInvitation handle incoming invitation
+func (c *Client) HandleInvitation(payload []byte) error {
+	// get the message type from the payload
+	msgType := &struct {
+		Type string `json:"@type,omitempty"`
+	}{}
+	err := json.Unmarshal(payload, msgType)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal handle invitation payload: %w", err)
+	}
+	if err = c.didexchangeSvc.Handle(dispatcher.DIDCommMsg{Type: msgType.Type, Payload: payload}); err != nil {
```

Comment:
May be we can add a validation to check for the message type.

Reply:
done

Reply:
Shall we add todo that in future we need service handle to send back connection ID  in addition to error ?